AG-5670 Bullet themes and defaults

olegat opened the pull request:
https://ag-grid.atlassian.net/browse/AG-9301
https://ag-grid.atlassian.net/browse/AG-9321
https://ag-grid.atlassian.net/browse/AG-9323

- Fixes incorrect param&return types for `customDefaultAxesSwapper` https://github.com/ag-grid/ag-charts/blob/83422ce66b91c212454f6d89f25410fc8a25ff07/packages/ag-charts-community/src/module/coreModules.ts#L101
- Adds files _bulletDefaults.ts_ and _bulletThemes.ts_.
- Adds styling options to `BulletSeries`
- Adds `scale.max` property to `BulletSeries`. Clip the colour ranges to not exceed `scale.max`

Test file:
https://plnkr.co/edit/BqtSjxH0R7fgHi6g?open=main.js

olegat committed:
AG-9324 Fix customDefaultAxesSwapper param&return types

olegat committed:
AG-9321 Add bullet defaults/styling/themes

olegat committed:
AG-9301 Process only the first datum in bullet graphs

olegat committed:
AG-9321 Clip colorRange[].stop to scale.max

olegat committed:
AG-9301 Clip everything to scale.max

olegat committed:
AG-9301 Update AgBulletSeriesOptions

alantreadway commented on packages/ag-charts-community/src/options/series/cartesian/bulletOptions.ts:
```diff
@@ -16,7 +17,17 @@ export interface AgBulletSeriesTooltipRendererParams<TDatum = any>
+
+export interface AgBulletScaleOptions {
+    /** Maximum value of the scale. Any values exceeding this number will be clipped to this maximum */
+    max?: number;
```
The way to achieve this with other chart-types is via the axis options - we should follow the same approach for this.

See https://ag-grid.com/javascript-charts/axes-domain/#domain-min--max

olegat replied:
I thought of that, but @AG-DavidG and I discussed this. We came to the conclusion that bullet graphs are gauges that don't conceptually have any axes (even though the underlying implementation uses axes). So having a dedicated `scale.max` makes sense in this case.

Although we could perhaps just have `scale.max` be an alias for `axes[0].domain.max` to make it easier to maintain.

olegat committed:
AG-9301 Use axes[0].max in bullet series

Keep the series[0].scale.max property, but make this an alias that
the default (axes[0].max = undefined).

alantreadway reviewed (approved):
LGTM
